app/actions/nodeNetworkActions.js

```diff
@@ -10,67 +11,60 @@ import {
 } from '../core/constants'
 
 const ID = 'nodeNetwork'
+const MAX_RESPONSE_TIME = 10000
+const RPC_PING_OVERRIDE = 10000
 
-const getBlockCount = async node =>
-  new Promise(resolve => {
-    let url = node.protocol ? `${node.protocol}://${node.url}` : node.url
-    url = node.port ? `${url}:${node.port}` : url
+const pingNodes = (nodes: Array<any>) => {
+  settings.timeout.ping = RPC_PING_OVERRIDE
+  const responses = []
+  return new Promise(resolve => {
+    nodes.forEach(node => {
+      let url = node.protocol ? `${node.protocol}://${node.url}` : node.url
+      url = node.port ? `${url}:${node.port}` : url
 
-    const client = new rpc.RPCClient(url)
-    client
-      .ping()
-      .then(latency => {
-        if (client.lastSeenHeight !== 0) {
-          resolve({
-            url,
-            blockCount: client.lastSeenHeight,
-            latency,
-          })
-        }
-      })
-      .catch({})
-  })
+      const client = new rpc.RPCClient(url)
+      client
+        .ping()
+        .then(latency => {
+          if (client.lastSeenHeight !== 0) {
+            responses.push({
+              url,
+              blockCount: client.lastSeenHeight,
+              latency,
+            })
+          }
+          if (responses.length === nodes.length) {
+            resolve(responses)
+          }
+        })
+        .catch(console.error)
 
-const raceNodePromises = (total, promises) => {
-  const responses = []
-  return new Promise(resolve =>
-    promises.forEach(promise =>
-      promise.then(result => {
-        responses.push(result)
-        if (responses.length === total) resolve(responses)
-      }),
-    ),
-  )
+      // if condition on line 58 is never met return the results
```

Comment:
I think its kinda problematic specifying a line number because it may change quickly making this comment irrelevant... Also, its always good to clear the timeout when you got the desired the result.